operators/src/source/gdal_source/mod.rs

```diff
@@ -1616,4 +1752,141 @@ mod tests {
             dataset_parameters.gdal_config_options,
         );
     }
+
+    #[test]
+    fn gdal_geotransform_to_bounds_neg_y_0() {
+        let gt = GdalDatasetGeoTransform {
+            origin_coordinate: Coordinate2D::new(0., 0.),
+            x_pixel_size: 1.,
+            y_pixel_size: -1.,
+        };
+
+        let sb = gt.standardized_spatial_partition(10, 10);
+
+        let exp = SpatialPartition2D::new(Coordinate2D::new(0., 0.), Coordinate2D::new(10., -10.))
+            .unwrap();
+
+        assert_eq!(sb, exp);
+    }
+
+    #[test]
+    fn gdal_geotransform_to_bounds_neg_y_5() {
+        let gt = GdalDatasetGeoTransform {
+            origin_coordinate: Coordinate2D::new(5., 5.),
+            x_pixel_size: 0.5,
+            y_pixel_size: -0.5,
+        };
+
+        let sb = gt.standardized_spatial_partition(10, 10);
+
+        let exp =
+            SpatialPartition2D::new(Coordinate2D::new(5., 5.), Coordinate2D::new(10., 0.)).unwrap();
+
+        assert_eq!(sb, exp);
+    }
+
+    #[test]
+    fn gdal_geotransform_to_bounds_pos_y_0() {
+        let gt = GdalDatasetGeoTransform {
+            origin_coordinate: Coordinate2D::new(0., 0.),
+            x_pixel_size: 1.,
+            y_pixel_size: 1.,
+        };
+
+        let sb = gt.standardized_spatial_partition(10, 10);
+
+        let exp = SpatialPartition2D::new(Coordinate2D::new(0., 10.), Coordinate2D::new(10., 0.))
+            .unwrap();
+
+        assert_eq!(sb, exp);
+    }
+
+    #[test]
+    fn gdal_geotransform_to_bounds_pos_y_5() {
+        let gt = GdalDatasetGeoTransform {
+            origin_coordinate: Coordinate2D::new(5., -5.),
+            x_pixel_size: 0.5,
+            y_pixel_size: 0.5,
+        };
+
+        let sb = gt.standardized_spatial_partition(10, 10);
+
+        let exp = SpatialPartition2D::new(Coordinate2D::new(5., 0.), Coordinate2D::new(10., -5.))
+            .unwrap();
+
+        assert_eq!(sb, exp);
+    }
+
+    #[test]
+    fn gdal_read_window_a() {
```

Comment:
I guess we need a test that reads from a flipped tiff in order to test if it all actually works

Reply:
[feb6a53](https://github.com/geo-engine/geoengine/pull/602/commits/feb6a534eb0c19d7b3f55e8c6714186ac1e17150)